TreeGrid.php

```diff
@@ -375,5 +382,25 @@ protected function guessColumns()
             }
         }
     }
+    
+    /**
+     * build tree
+     * @param array $elements
+     * @param string $parentId
+     * @return array
+     */
+    protected  function buildTree(array $elements, $parentId = null) {
+        $branch = [];
+        foreach ($elements as $key => $element) {
+            if ($element[$this->parentColumnName] == $parentId) {
+                $children = $this->buildTree($elements, $element[$this->keyColumnName]);
+                $branch[$key] = $element;
+                if ($children) {
+                    $branch = $branch + $children;
```

Comment:
Added recursion to find sub classes.  
Recursion from $parentRoot  
Please see http://maxazan.github.io/jquery-treegrid/  step2